Telemetry/TelemetryScheduler.swift

```diff
@@ -21,15 +21,25 @@ public class TelemetryScheduler {
     }
     
     public func scheduleUpload(pingType: String, completionHandler: @escaping () -> Void) {
-        guard var pings = storage.read(pingType: pingType) else {
-            completionHandler()
-            return
-        }
+        var pingSequence = storage.sequenceForPingType(pingType)
+
+        func uploadNextPing() {
+            guard !hasReachedDailyUploadLimitForPingType(pingType) else {
+                completionHandler()
+                return
+            }
+
+            // Get the next ping in the sequence.
+            var nextPing = pingSequence.next()
+
+            // If there are no remaining pings in the sequence, get a new sequence from
+            // storage in case any additional pings were queued while we were uploading.
+            if nextPing == nil {
```

Comment:
next line should do `pingSequence.remove()` in the case of nil nextPing;  that would make sense to me at least because the sequence iterator loops over files in a dir and the file may not be convertible to a TelemetryPing.
Also good to add that to the comment above

Reply:
(ignore see other remove() comment)